test/test-cases/functional/saidashvnet.py

```diff
@@ -0,0 +1,763 @@
+
+    def pa_validation_create(self, sip, vnet_id):
+        """
+        Create source PA validation entry
+        """
+
+        pa_validation_entry = sai_thrift_pa_validation_entry_t(switch_id=self.switch_id,
+                                                               sip=sai_ipaddress(sip),
+                                                               vnet_id=vnet_id)
+        sai_thrift_create_pa_validation_entry(self.client,
+                                              pa_validation_entry,
+                                              action=SAI_PA_VALIDATION_ENTRY_ACTION_PERMIT)
+        self.assertEqual(self.status(), SAI_STATUS_SUCCESS)
+        self.add_teardown_obj(self.pa_validation_remove, pa_validation_entry)
+
+    def pa_validation_remove(self, pa_validation_entry):
+        sai_thrift_remove_pa_validation_entry(self.client, pa_validation_entry)
+
+    def outbound_routing_vnet_direct_create(self, eni_id, lpm, dst_vnet_id, overlay_ip):
+        """
+        Create outband vnet direct routing entry
+        """
+
+        outbound_routing_entry = sai_thrift_outbound_routing_entry_t(
+                                switch_id=self.switch_id, eni_id=eni_id,
+                                destination=sai_ipprefix(lpm))
+        sai_thrift_create_outbound_routing_entry(self.client, outbound_routing_entry,
+                                                 dst_vnet_id=dst_vnet_id, overlay_ip=sai_ipaddress(overlay_ip),
+                                                 action=SAI_OUTBOUND_ROUTING_ENTRY_ACTION_ROUTE_VNET_DIRECT)
+        self.assertEqual(self.status(), SAI_STATUS_SUCCESS)
+        self.add_teardown_obj(self.outbound_routing_vnet_direct_remove, outbound_routing_entry)
+
+    def outbound_routing_direct_create(self, eni_id, lpm):
+        """
+        Create outband vnet direct routing entry
+        """
+
+        outbound_routing_entry = sai_thrift_outbound_routing_entry_t(
+                                switch_id=self.switch_id, eni_id=eni_id,
+                                destination=sai_ipprefix(lpm))
+        sai_thrift_create_outbound_routing_entry(self.client, outbound_routing_entry,
+                                                 action=SAI_OUTBOUND_ROUTING_ENTRY_ACTION_ROUTE_DIRECT)
+        self.assertEqual(self.status(), SAI_STATUS_SUCCESS)
+        self.add_teardown_obj(self.outbound_routing_vnet_direct_remove, outbound_routing_entry)
+
+    def outbound_routing_vnet_direct_remove(self, entry):
+        sai_thrift_remove_outbound_routing_entry(self.client, entry)
+
+    def outbound_ca_to_pa_create(self, dst_vnet_id, dip, underlay_dip):
+        """
+        Create outband CA PA mapping
+        """
+
+        ca_to_pa_entry = sai_thrift_outbound_ca_to_pa_entry_t(switch_id=self.switch_id,
+                                                              dst_vnet_id=dst_vnet_id, dip=sai_ipaddress(dip))
+        sai_thrift_create_outbound_ca_to_pa_entry(self.client, ca_to_pa_entry,

```

Comment:
@vmytnykx 
as per p4 outbound code, the overlay dmac is always provided as part of the ca to pa mapping data.

 action set_tunnel_mapping(IPv4Address underlay_dip,
                              EthernetAddress overlay_dmac,
                              bit<1> use_dst_vnet_vni) {
        if (use_dst_vnet_vni == 1)
            meta.vnet_id = meta.dst_vnet_id;
        meta.encap_data.overlay_dmac = overlay_dmac;
        meta.encap_data.underlay_dip = underlay_dip;
    }

any reason the dmac is not provided in those testcases?